config/initializers/define-orm-models.ts

```diff
@@ -14,14 +14,14 @@ export default {
    * any internal model representation necessary.
    */
   async initialize(application: Application): Promise<void> {
-    let container = application.container;
     let models: { [modelName: string]: typeof Model } = application.container.lookupAll('model');
     let modelsGroupedByAdapter = new Map();
-    forEach(models, (ModelClass, modelName) => {
-      if (ModelClass.abstract) {
+    forEach(models, (ModelClass: typeof Model) => {
+      if (ModelClass.hasOwnProperty('abstract') && ModelClass.abstract) {
         return;
       }
-      let Adapter = container.lookup(`orm-adapter:${ modelName }`, { loose: true }) || container.lookup('orm-adapter:application');
+      // Let the model class resolve its own adapter
+      let Adapter = ModelClass.getAdapter(application.container);
```

Comment:
Not sure about this - especially if we move the model query methods to the database service (as discussed).

I see some value in letting a Model class determine it's own adapter, and the rest of the system respecting that. But on the other hand, is that level of customization really useful? It kinda feels like it's defeating the whole point of the container's conventions.

I'm on the fence, curious to hear what others think.